src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/android/AndroidBrokerProxy.cs

```diff
@@ -226,21 +240,48 @@ public Intent GetIntentForBrokerActivity(AuthenticationRequest request, Activity
                 // logic
                 if (intent != null)
                 {
+                    _logger.Info("BrokerProxy: Intent created from BundleResult is not null. ");
                     intent.PutExtra(BrokerConstants.BrokerRequest, BrokerConstants.BrokerRequest);
                 }
+
+                _logger.Info("BrokerProxy: Intent created from BundleResult is null. ");
             }
-            catch (AdalException)
+            catch (AdalException ex)
             {
-                throw;
+                _logger.ErrorPii(ex);
+                throw new AdalException(
+                    AdalError.Unknown,
+                    ex.Message);
             }
             catch (Exception e)
             {
                 _logger.ErrorPii(e);
```

Comment:
>lo [](start = 17, length = 2)

as above. do we log anything even if pii is not enabled? we should to ensure the log says something bad happened.

Reply:
yes..i think we do log it.

---
In reply to: [286275005](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1598#discussion_r286275005) [](ancestors = 286275005)